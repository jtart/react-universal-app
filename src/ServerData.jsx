import React from 'react';

const ServerData = ({ id, data }) => {
  const serialisedProps = JSON.stringify(data).replace(/</g, '\\u003c');

  return (
    <script
      id={id}
      type="application/json"
      dangerouslySetInnerHTML={{
        __html: serialisedProps,
      }}
    />
  );
};

export default ServerData;
