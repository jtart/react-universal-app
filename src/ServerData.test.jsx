import React from 'react';
import ServerData from './ServerData';

describe('ServerData', () => {
  it('should a render script with serialised props', () => {
    const wrapper = shallow(
      <ServerData
        id="idOne"
        data={{ data: { entry: 'This is some data!' } }}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
