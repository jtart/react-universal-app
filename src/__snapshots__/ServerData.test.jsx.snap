// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServerData should a render script with serialised props 1`] = `
<script
  dangerouslySetInnerHTML={
    Object {
      "__html": "{\\"data\\":{\\"entry\\":\\"This is some data!\\"}}",
    }
  }
  id="idOne"
  type="application/json"
/>
`;
